File: src/main/java/ru/kemsu/openenv/service/BasicOrganisationService.java
package ru.kemsu.openenv.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import ru.kemsu.openenv.model.Organisation;
import ru.kemsu.openenv.repository.OrganisationRepository;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class BasicOrganisationService implements OrganisationService {

    private final OrganisationRepository repository;

    @Autowired
    public BasicOrganisationService(final OrganisationRepository repository) {
        this.repository = repository;
    }



    @Override
    public Organisation create(final Organisation organisation) {
        organisation.setCreatedAt(String.valueOf(LocalDateTime.now()));
        return repository.save(organisation);
    }

    @Override
    public Organisation find(String id) {
        return repository.findById(id).get();
    }

    @Override
    public Organisation findByName(final String organisationName) {
        return repository.findByName(organisationName);
    }


    @Override
    public List<Organisation> findAll() {
        return repository.findAll();
    }

    @Override
    public Organisation update(final String id, final Organisation organisation) {
        organisation.setId(id);

        final Organisation saved = repository.findById(id).get();

        if (saved != null) {
            organisation.setCreatedAt(saved.getCreatedAt());
            organisation.setUpdatedAt(String.valueOf(LocalDateTime.now()));
        } else {
            organisation.setCreatedAt(String.valueOf(LocalDateTime.now()));
        }
        repository.save(organisation);
        return organisation;
    }

    @Override
    public String delete(final String id) {
        repository.deleteById(id);
        return id;
    }

}

File: src/main/java/ru/kemsu/openenv/service/BasicOrganisationService.kt
package ru.kemsu.openenv.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import ru.kemsu.openenv.model.Organisation
import ru.kemsu.openenv.repository.OrganisationRepository
import java.time.LocalDateTime

@Service
class BasicOrganisationService @Autowired constructor(private val repository: OrganisationRepository) : OrganisationService {
    override fun create(organisation: Organisation): Organisation {
        organisation.createdAt = LocalDateTime.now().toString()
        return repository.save(organisation)
    }

    override fun find(id: String): Organisation {
        return repository.findById(id).get()
    }

    override fun findByName(organisationName: String): Organisation {
        return repository.findByName(organisationName)
    }

    override fun findAll(): List<Organisation> {
        return repository.findAll()
    }

    override fun update(id: String, organisation: Organisation): Organisation {
        organisation.id = id
        val saved = repository.findById(id).get()
        if (saved != null) {
            organisation.createdAt = saved.createdAt
            organisation.updatedAt = LocalDateTime.now().toString()
        } else {
            organisation.createdAt = LocalDateTime.now().toString()
        }
        repository.save(organisation)
        return organisation
    }

    override fun delete(id: String): String {
        repository.deleteById(id)
        return id
    }

}
